trace/cloud-trace-demo-app-opentelemetry/app/app.py

```diff
@@ -11,49 +11,71 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
 """
 A sample app demonstrating CloudTraceSpanExporter
 """
+
+import os
 import random
 import time
 
-# [START trace_demo_imports]
 import flask
-from opentelemetry import propagate, trace
+# [START trace_demo_imports]
+from opentelemetry import trace
 from opentelemetry.exporter.cloud_trace import CloudTraceSpanExporter
 from opentelemetry.instrumentation.flask import FlaskInstrumentor
+from opentelemetry.instrumentation.requests import RequestsInstrumentor
+from opentelemetry.propagate import set_global_textmap
 from opentelemetry.propagators.cloud_trace_propagator import CloudTraceFormatPropagator
 from opentelemetry.sdk.trace import TracerProvider
-from opentelemetry.sdk.trace.export import SimpleSpanProcessor
-
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
 # [END trace_demo_imports]
-
-app = flask.Flask(__name__)
-FlaskInstrumentor().instrument_app(app)
+import requests
 
 
+# [START trace_demo_create_exporter]
 def configure_exporter(exporter):
-    trace.set_tracer_provider(TracerProvider())
-    trace.get_tracer_provider().add_span_processor(SimpleSpanProcessor(exporter))
-    propagate.set_global_textmap(CloudTraceFormatPropagator())
+    set_global_textmap(CloudTraceFormatPropagator())
+    tracer_provider = TracerProvider()
+    tracer_provider.add_span_processor(BatchSpanProcessor(exporter))
+    trace.set_tracer_provider(tracer_provider)
 
 
 configure_exporter(CloudTraceSpanExporter())
 tracer = trace.get_tracer(__name__)
+# [END trace_demo_create_exporter]
+
+
+# [START trace_demo_middleware]
+app = flask.Flask(__name__)
+FlaskInstrumentor().instrument_app(app)
+RequestsInstrumentor().instrument()
+# [END trace_demo_middleware]
 
 
 @app.route("/")
 def template_test():
     # Sleep for a random time to imitate a random processing time
     time.sleep(random.uniform(0, 0.5))
 
-    with tracer.start_as_current_span("span1"):
-        with tracer.start_as_current_span("span2"):
-            with tracer.start_as_current_span("span3"):
-                print("Hello world from Cloud Trace Exporter!")
-
-    return "Hello World"
+    # If there is an endpoint, send keyword to next service.
+    # Return received input with the keyword
+    keyword = os.getenv("KEYWORD")
+    endpoint = os.getenv("ENDPOINT")
+# [START trace_context_header]
+    if endpoint is not None and endpoint != "":
+        data = {'body': keyword}
+        response = requests.get(
+            endpoint,
+            params=data
```

Comment:
```suggestion
            params=data,
```

Reply:
I think it might introduce a confusion to some readers. Especially, if it is done for only one function call.

Reply:
This is a common Python style (https://peps.python.org/pep-0008/#when-to-use-trailing-commas) and is strongly encouraged when you list out items vertically rather than horizontally.

Reply:
Understood. Thank you for the link to the guidelines.